### DISCUSSION TITLE
Raspberry Pi Pico ADC changes the sampling rate in Free-running Sampling mode

### DISCUSSION THREAD

**Discussion by leezisheng** [micropython/micropython#15815] [Hardware & Peripherals]
I am configuring the ADC sampling rate here. Due to unknown reasons, the DIV register cannot be configured here. The actual sampling period is fixed at 65.
Can anyone help me what is going wrong here？

<img width="512" alt="b078a23843d708c1ae0b4c6966c8a88" src="https://github.com/user-attachments/assets/93323e84-5b2a-49e5-a9d4-db18c5821c0d">
<img width="554" alt="a91a38f1944c012e122d14c44dcbc40" src="https://github.com/user-attachments/assets/8377890b-4583-4a30-8977-6207108fd8c9">
<img width="551" alt="c39121a7385513c1579481977b0c62e" src="https://github.com/user-attachments/assets/49c49bfc-7551-45d2-ac80-b89daa824a3a">

**Comment by scruss** [comment]
The existing bits in `div_value` are overwriting the bits you want to set from `int_part` and `frac_part`. Instead of
```
div_value | (int_part << 8) | frac_part
```
try
```
(div_value & 0xff000000) | (int_part << 8) | frac_part
```
